src/domain/auth-rebac/domain/src/repos/rebac_repository.rs

```diff
@@ -0,0 +1,378 @@
+        Self {
+            entity_type,
+            entity_id: entity_id.into(),
+        }
+    }
+
+    pub fn new_account(entity_id: impl Into<Cow<'a, str>>) -> Self {
+        Self::new(EntityType::Account, entity_id)
+    }
+
+    pub fn new_dataset(entity_id: impl Into<Cow<'a, str>>) -> Self {
+        Self::new(EntityType::Dataset, entity_id)
+    }
+
+    pub fn into_owned(self) -> Entity<'static> {
+        Entity {
+            entity_type: self.entity_type,
+            entity_id: Cow::Owned(self.entity_id.into_owned()),
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct EntityWithRelation<'a> {
+    pub entity: Entity<'a>,
+    pub relation: Relation,
+}
+
+impl<'a> EntityWithRelation<'a> {
+    pub fn new(entity: Entity<'a>, relation: Relation) -> Self {
+        Self { entity, relation }
+    }
+
+    pub fn new_account(entity_id: impl Into<Cow<'a, str>>, relation: Relation) -> Self {
+        let account_entity = Entity::new_account(entity_id);
+
+        Self::new(account_entity, relation)
+    }
+
+    pub fn new_dataset(entity_id: impl Into<Cow<'a, str>>, relation: Relation) -> Self {
+        let dataset_entity = Entity::new_dataset(entity_id);
+
+        Self::new(dataset_entity, relation)
+    }
+}
+
+pub type ObjectEntity<'a> = Entity<'a>;
+pub type ObjectEntityWithRelation<'a> = EntityWithRelation<'a>;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub type PropertyValue<'a> = Cow<'a, str>;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum PropertyName {
+    Dataset(DatasetPropertyName),
+    Account(AccountPropertyName),
+}
+
+impl PropertyName {
+    pub fn dataset_allows_anonymous_read<'a>(allows: bool) -> (Self, PropertyValue<'a>) {
+        let value = if allows { "true" } else { "false" };
+
+        (
+            Self::Dataset(DatasetPropertyName::AllowsAnonymousRead),
+            value.into(),
+        )
+    }
+
+    pub fn dataset_allows_public_read<'a>(allows: bool) -> (Self, PropertyValue<'a>) {
+        let value = if allows { "true" } else { "false" };
+
+        (
+            Self::Dataset(DatasetPropertyName::AllowsPublicRead),
+            value.into(),
+        )
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum DatasetPropertyName {
+    AllowsAnonymousRead,
+    AllowsPublicRead,
+}
+
+impl From<DatasetPropertyName> for PropertyName {
+    fn from(value: DatasetPropertyName) -> PropertyName {
+        PropertyName::Dataset(value)
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum AccountPropertyName {
+    // TBA
```

Comment:
This could be "admin" indication

Reply:
Yes, agree -- but for now, it's a feature of the accounts domain

Added a task in order to make this change:
- https://github.com/kamu-data/kamu-cli/issues/766